File: patterns/cs/src/RefactoringGuru/AbstractFactory/Example/Factories/MacOSFactory.cs
﻿using RefactoringGuru.AbstractFactory.Example.Buttons;
using RefactoringGuru.AbstractFactory.Example.Checkboxes;

namespace RefactoringGuru.AbstractFactory.Example.Factories
{
    /**
     * EN: Each concrete factory extends basic factory and responsible for creating
     * products of a single variety.
     * 
     * RU: Каждая конкретная фабрика знает и создаёт только продукты своей вариации.
     */
    class MacOSFactory : IGuiFactory
    {
        public IButton CreateButton()
        {
            return new MacOsButton();
        }

        public ICheckbox CreateCheckbox()
        {
            return new MacOsCheckbox();
        }
    }
}

File: patterns/php/src/AbstractFactory/Example/Factories/MacOSFactory.php
<?php

namespace RefactoringGuru\Patterns\AbstractFactory\Example\Factories;

use RefactoringGuru\Patterns\AbstractFactory\Example\Buttons\MacOSButton;
use RefactoringGuru\Patterns\AbstractFactory\Example\Checkboxes\MacOSCheckbox;

class MacOSFactory implements GUIFactoryInterface
{
    /**
     * @inheritdoc
     */
    public function createButton()
    {
        return new MacOSButton();
    }

    /**
     * @inheritdoc
     */
    public function createCheckbox()
    {
        return new MacOSCheckbox();
    }
}
